git-sync/main.go

```diff
@@ -127,19 +134,28 @@ func main() {
 }
 
 // syncRepo syncs the branch of a given repository to the destination at the given rev.
-func syncRepo(repo, dest, branch, rev string, depth int) error {
-	gitRepoPath := path.Join(dest, ".git")
+func syncRepo(repo, dest, branch, rev, subdir string, depth int, atomic bool) error {
+	pullTarget := path.Join(dest, subdir)
+	gitRepoPath := path.Join(pullTarget, ".git")
 	_, err := os.Stat(gitRepoPath)
 	switch {
 	case os.IsNotExist(err):
+		if atomic {
```

Comment:
If the repo does exist, then we've already done our initial pull, so we don't need to pull into the tempdir. We just pull into the existing repo. This PR only covers making the first pull atomic.